Sources/Sentry/SentryCrashReportConverter.m

```diff
@@ -46,7 +46,21 @@ - (instancetype)initWithReport:(NSDictionary *)report
 
         self.diagnosis = crashContext[@"diagnosis"];
         self.exceptionContext = crashContext[@"error"];
-        self.threads = crashContext[@"threads"];
+        [self initThreads:crashContext[@"threads"]];
+    }
+    return self;
+}
+
+- (void)initThreads:(NSArray<NSDictionary *> *)threads
+{
+    if (nil != threads && [threads isKindOfClass:[NSArray class]]) {
+        // SentryCrash sometimes produces recrash_reports where an element of threads is a
+        // NSString instead of a NSDictionary. When this happens we can't read the details of
+        // the thread, but we have to discard it. Otherwise we would crash.
```

Comment:
M: I am wondering if actually we should have fixed SentryCrash instead? if SentryCrash writes a String instead of an Array which is probably a bug, this `isKindOfClass` guards on the type but if SentryCrash also writes a broken Array? I'd doublecheck SentryCrash's code too and if it's not a valid Array-threads list, just set it to null, wdyt?

Reply:
I double-checked and couldn't find the cause of this problem. It could also be that somebody else modified the JSON. The real problem behind this is, that if the JSON gets modified in a way we don't expect it we always crash. To avoid those crashes we would always need to check if the JSON is as we expect it to be. Do you think it's worth doing so @marandaneto, @HazAT and @bruno-garcia ?

Reply:
I don't think so, if somebody modifies the json it'd break on android too, only the parsing though, there's a higher level try/catch that would discard the whole file altogether, maybe we do something like this at least? if parsing failed, we delete it.

Reply:
> If parsing failed, we delete it. 

This sounds reasonable. We def don't want to crash because we couldn't parse something. Better discard the crash file and move on. It's likely a bug we introduced by not handling a file we serialized on an older version

Reply:
I added a try-catch and a test to make sure the report is also deleted.